src/js/stores/NewsFeedStore.jsx

```diff
@@ -78,22 +84,21 @@ class NewsFeedStore extends EventEmitter {
       failure: function(error){
         dispatcher.dispatch({type: "NEWSFEED_ERROR", error: error});
       }
-
     });
-
   }
 
 
   handleActions(action) {
     switch(action.type) {
       case "NEWSFEED_POST": {
-        // TODO: Readd this back in
-        this.emit("updated");
+        // TODO: ensure that client sees their post on the front end
+        this.emit("post");
         break;
       }
       case "NEWSFEED_GET": {
-        this.posts = action.posts;
-        // TODO: Append posts, don't simply delete old data
+        Array.prototype.push.apply(this.contentFeed.items, action.response.data.posts[0]);
```

Comment:
I don't know how the back end works, but this seems potentially sketchy. Will the get ever return more than one post?